webpay/pay/tasks.py

```diff
@@ -105,6 +102,10 @@ def configure_transaction(request, trans=None, mcc=None, mnc=None):
                     request.session['uuid'],
                     [p.name for p in providers])
 
+    # Now that the background task has been started successfully,
+    # prevent configuration from running twice.
+    request.session['configured_trans'] = request.session['trans_id']
```

Comment:
no, it doesn't block. But it *might* raise an error if it can't connect to the celery daemon to queue up the task. My patch will make it so that the client attempts to reconfigure the transaction in the case of a celery connection failure (which might be intermittent).